.github/workflows/ci.yml

```diff
@@ -1,24 +1,58 @@
 name: CI
 
+concurrency:
+  group: "${{github.workflow}}-${{github.ref}}"
+  cancel-in-progress: true
+
 on:
   push:
-    branches:
-      - main
+    branches: [ main ]
   pull_request:
+    types: [opened, synchronize]
+    branches: [ '*' ]
 
 jobs:
   build:
     name: Browser tests
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
       - uses: actions/setup-node@v3
         with:
           node-version: 16
           cache: "yarn"
       - name: Install Dependencies
         run: yarn install --frozen-lockfile
       - run: bin/ci
+  rails-tests:
+    name: Downstream Rails integration tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: "yarn"
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: "3.4"
+      - name: Install Dependencies
+        run: yarn install --frozen-lockfile
+      - name: Packaging
+        run: yarn build
+      - name: Clone Rails
+        run: git clone --depth=1 https://github.com/rails/rails
+      - name: Configure Rails
+        run: |
+          sudo apt install libvips-tools
+          cd rails
+          yarn install --frozen-lockfile
+          bundle add action_text-trix --path ".."
+          bundle show --paths action_text-trix
+      - name: Action Text tests
+        run: |
+          cd rails/actiontext
+          bundle exec rake test test:system
```

Comment:
This is exciting! Nice work.

Were there any lessons learned in this process that could be applied to `rails/rails`? From what I recall, [buildkite isn't currently executing the System Test suite for Action Text as part of CI runs](https://github.com/rails/rails/pull/46807#issuecomment-1793312885).

Reply:
@seanpdoyle I don't think I have anything to add to those threads, which seem related at various points to Buildkite, container networking, karma browser coverage, and selenium configuration. I don't know much about any of those things!

By default, Action Text's system tests will use headless chrome, which is [installed on the Github Actions runner](https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2404-Readme.md#browsers-and-drivers) so it Just Worked™ for me for this limited use case.